providers/implementations/ciphers/cipher_sm4_xts.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <crypto/sm4.h>
+#include "prov/ciphercommon.h"
+#include "crypto/sm4_platform.h"
+
+
+PROV_CIPHER_FUNC(void, xts_stream,
+                 (const unsigned char *in, unsigned char *out, size_t len,
+                  const SM4_KEY *key1, const SM4_KEY *key2,
+                  const unsigned char iv[16]));
+
+typedef struct prov_sm4_xts_ctx_st {
+    PROV_CIPHER_CTX base; /* Must be first */
+    union {
+        OSSL_UNION_ALIGN;
+        SM4_KEY ks;
+    } ks1, ks2; /* SM4 key schedules to use */
+    int xts_standard; /* Must be 0 or 1, 0 for XTS mode specified by GB/T 17964-2021, 1
+                for XTS mode specified by IEEE Std 1619-2007 */
+    XTS128_CONTEXT xts;
+    OSSL_xts_stream_fn stream_gb; /* Stream function for XTS mode specified by
```

Comment:
> 

The OSSL_xts_stream_fn is set in EVP_CipherInit_ex, while the xts_standard is determined in EVP_CIPHER_CTX_set_params.

I think there must be two OSSL_xts_stream_fns for two xts_standards respectively, since the EVP_CIPHER_CTX_set_params need to be used after EVP_CipherInit_ex.

Reply:
> this should be a union which include stream_gb and stream since xts_standard determine which steam_fn to use.
union cannot be used for the same reason, we need two memory locations for both stream_gb and stream, and determine which to use later.